tests/integration/ha/test_horizontal_scaling.py

```diff
@@ -396,5 +398,16 @@ async def test_safe_scale_down_remove_leaders(
     cluster_health_resp = await cluster_health(ops_test, leader_unit_ip, wait_for_green_first=True)
     assert cluster_health_resp["status"] == "green"
 
+    # remove node containing primary shard of index "series_index"
+    shards = await get_shards_by_index(ops_test, leader_unit_ip, ContinuousWrites.INDEX_NAME)
+    unit_with_primary_shard = [shard.unit_id for shard in shards if shard.is_prim][0]
+    await ops_test.model.applications[app].destroy_unit(f"{app}/{unit_with_primary_shard}")
+
+    shards = await get_shards_by_index(ops_test, leader_unit_ip, ContinuousWrites.INDEX_NAME)
```

Comment:
it is possible that the unit you removed on line 404 was the leader - so as a best practice re-caulcate the leader after waiting for a new one to be elected
```suggestion
    await ops_test.model.wait_for_idle(
        apps=[app],
        status="active",
        timeout=1000,
        wait_for_exact_units=init_units_count-1,
        idle_period=IDLE_PERIOD,
    )
    leader_unit_ip = await get_leader_unit_ip(ops_test, app=app)
    shards = await get_shards_by_index(ops_test, leader_unit_ip, ContinuousWrites.INDEX_NAME)
```

Reply:
good catch! thank you